test/production/required-server-files.test.ts

```diff
@@ -980,6 +984,35 @@ describe('should set-up next', () => {
     expect(JSON.parse($('#router').text()).asPath).toBe('/an-ssg-path')
   })
 
+  it('should have correct asPath on dynamic SSG page fallback correctly', async () => {
+    const toCheck = [
+      {
+        pathname: '/fallback-false/first',
+        matchedPath: '/fallback-false/first',
+      },
+      {
+        pathname: '/fallback-false/first',
+        matchedPath: `/_next/data/${next.buildId}/fallback-false/first.json`,
+      },
+    ]
+    for (const check of toCheck) {
+      console.warn('checking', check)
```

Comment:
You can use `test.each()` instead

https://stackoverflow.com/a/53211296/266535